### DISCUSSION TITLE
Better handling of "MISSING UNIT IMAGE"?

### DISCUSSION THREAD

**Discussion by asvitkine** [triplea-game/triplea#10138] [General]
Hello!

If we look at the list of issues reported to the project, there's tons of "MISSING UNIT IMAGE" reports.

My understanding is such issues are mainly problems with maps themselves, rather than the engine. So it seems the triplea-game/triplea issue tracker is not the best place for these!

So one option would be to report them to the actual maps being played.

However, perhaps there's an even simpler solution? Why not just have a placeholder image that's used when the unit image is missing? This way, the error goes away entirely and for cases where it causes problems for the map, people can report those specific visual display problems to the map's maintainers separately?

This would help clean out the spammy reports from the issue database.

**Comment by RoiEXLab** [comment]
I agree, the missing unit images should be dealt with somehow (I think the engine provides some sort of fallback anyways) and an issue on this repository doesn't really help fixing the problem because lots of maps are not well-maintained.

Opening an issue for individual maps is not that easy because maps are not really connected to their repository in any meaningful way. It just happens that the engine downloads the  maps from the repository

**Comment by DanVanAtta** [comment]
A no-image image would be excellent. That would be a feature. This does not fully resolve the error.

Anything logged as a 'warning' will not be reportable. We can downgrade the error if we so choose. Presumably missing images makes the map relatively unplayable. Anther concern is that these issues would go unnoticed entirely.

There has been a suggestion to do something like attach an 'error-id' to error messages so that they can be de-duped more accurately. If we did this, we would instead of having a new issue be added, a new comment be added indicating the new details.

There has been other suggestions to help tackle the root cause of this, notably to create a 'map linter' that could detect any missing images. This could go further and be incorporated perhaps into the maps server which could then server a status page maybe that shows any map problems. Having at least a standalone tool to determine if a map  has all valid images would be a great item and would let us fully find all such cases and then fix them.

Last, the "3.0" initiative would likely seek to update the XML such that all image paths would be explicit. This would make finding missing images far more trivial.

Until then, we are in a bit of a place of what is anyone willing to do. The 3.0 approach seems like the best long term solution.

Of note, when 2.6 is launched, we would close all 2.5 error reports and that should help quite a bit. It would also be nice to see the server add a '2.5', '2.6' or a 'prerelease' label to error reports as well to make them easier to classify.

**Reply by DanVanAtta** [reply]
A no-image image is now implemented in 2.6

**Comment by jdimeo** [comment]
Our map throws this error (but rarely and non-determinstically) and not because of a bug in our map (and doesn't affect the playability). Probably because we're doing "munitions" that all get cleared out during phases and swapping one unit out for another, etc. etc. Weird stuff that confuses the engine sometimes. Just as a data point...

**Comment by DanVanAtta** [comment]
Sadly 2.6 is a bit stalled.. It already has a patch to show a placeholder
image instead of no image with error message.
